cdap-ui/app/cdap/components/Experiments/CreateView/SplitDataStep/index.js

```diff
@@ -37,94 +38,141 @@ const getSplitLogsUrl = (experimentId, splitInfo) => {
-  let isSplitComplete = ['Complete', 'Failed'].indexOf(splitStatus) !== -1;
-  let isSplitFailed = splitStatus === 'Failed';
-  const splitError = () => {
+const getSplitFailedElem = (experimentId, splitInfo) => {
+  return (
+    <span className="split-error-container">
+      Failed to split: Please check {" "}
+      <a href={getSplitLogsUrl(experimentId, splitInfo)} target="_blank"> Logs </a>{" "}
+      for more information
+    </span>
+  );
+};
+
+class SplitDataStep extends Component {
+  state = {
+    shouldShowSplitFailedAlert: false
+  };
+
+  static propTypes = {
+    splitInfo: PropTypes.object,
+    schema: PropTypes.object,
+    createSplitAndUpdateStatus: PropTypes.func,
+    setSplitFinalized: PropTypes.func,
+    setModelCreateError: PropTypes.func,
+    experimentId: PropTypes.string,
+    error: PropTypes.string
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.splitInfo.status && ['CREATING', 'Splitting'].indexOf(this.props.splitInfo.status) !== -1 && nextProps.splitInfo.status === 'Failed') {
+      this.setState({
+        shouldShowSplitFailedAlert: true
+      });
+    }
+  }
+
+  closeSplitFailedAlert = () => {
+    this.setState({
+      shouldShowSplitFailedAlert: false
+    });
+  }
+
+  renderSplitBtn() {
+    let isSplitCreated = Object.keys(this.props.splitInfo).length;
+    let splitStatus = (this.props.splitInfo || {}).status;
+    let isSplitComplete = ['Complete', 'Failed'].indexOf(splitStatus) !== -1;
+
+    if (!isSplitCreated || (isSplitCreated && isSplitComplete)) {
+      return (
+        <div>
+          <button
+            className="btn btn-primary"
+            onClick={this.props.createSplitAndUpdateStatus}
+          >
+            Split data Randomly and verify sample
+          </button>
+        </div>
+      );
+    }
     return (
-      <span className="split-error-container text-danger">
-        Current Split Failed. Please check {" "}
-        <a href={getSplitLogsUrl(experimentId, splitInfo)} target="_blank"> Logs </a>{" "}
-        for more information
-      </span>
+      <button
+        className="btn btn-primary"
+        disabled
+      >
+        <div className="btn-inner-container">
+          <IconSVG name="icon-spinner" className="fa-spin" />
+          <span>Splitting data</span>
+        </div>
+      </button>
     );
-  };
-  if (!isSplitCreated || (isSplitCreated && isSplitComplete) || (isSplitCreated && isSplitFailed)) {
+  }
+
+  render() {
+    let splitStatus = this.props.splitInfo.status || 'Complete';
+
     return (
-      <div>
-        <button
-          className="btn btn-primary"
-          onClick={onClick}
-        >
-          Split data Randomly and verify sample
-        </button>
-        {splitStatus === 'Failed' ? splitError() : null}
+      <div className="split-data-step">
+        <h3>Split Data </h3>
+        <div>Create test dataset for this model.</div>
+        <br />
+        {this.renderSplitBtn()}
+        {
```

Comment:
nit: Would be nice if we could split this into multiple render functions for easier readability.